### PR TITLE
feat: add tooltip to column header text which overflows

### LINKED ISSUE #10495: [Feature]: Make a change in the column headers such that the overflowing content is visible in the tooltip upon hover on column header.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

I would need tooltip for a column header since sometimes i have many columns and column name is big but i want to show as many columns as possible on view port. But this truncates table column name and i can't see the column header name without a tooltip. 

### Why should this be worked on?

This will allow users to have more columns visible on view port with context on column header name/title.

### PR DESCRIPTION
**Pull request by vicky-primathon**
## Description
Overflowing content is visible in the tooltip upon hover on column header.

Fixes #10495 

## Type of change

- New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: fix/table-header-text-overflow 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.72 **(0)** | 37.02 **(0.01)** | 35.18 **(0.01)** | 56.05 **(0)**
 :red_circle: | app/client/src/selectors/commentsSelectors.ts | 83.61 **(-1.64)** | 61.76 **(-2.95)** | 73.33 **(0)** | 88.24 **(-2.35)**
 :red_circle: | app/client/src/widgets/TableWidget/component/AutoToolTipComponent.tsx | 57.58 **(0)** | 38.71 **(-0.58)** | 33.33 **(0)** | 58.62 **(0)**
 :green_circle: | app/client/src/widgets/TableWidget/component/TableStyledWrappers.tsx | 87.5 **(0.6)** | 56.2 **(1.31)** | 80.7 **(1.07)** | 87.5 **(0.6)**</details>

### CODE REVIEW

File: app/client/src/widgets/TableWidget/component/TableStyledWrappers.tsx [hunk 1]
```diff
@@ -574,3 +574,16 @@ export const MenuCategoryWrapper = styled.div`
 export const MenuStyledOptionHeader = styled.div`
   font-weight: 600;
 `;
+
+export const ColumnTitle = styled.div`
```
Comment (sbalaji1192) [inline app/client/src/widgets/TableWidget/component/TableStyledWrappers.tsx:578]: Are we using ColumnTitle somewhere?

**Review by sbalaji1192 — APPROVED** [review]

**Comment by YogeshJayaseelan** [comment]
@vicky-primathon Tested this PR and tooltip are displayed on mouse hovering on the column header which has longer name

**Pull request merged**